test/integration/QueryRunnerTest.kt

```diff
@@ -0,0 +1,77 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+// We need to access the private function StudioState.client.session.tryOpen, this allows us to.
+@file:Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")
+
+package com.vaticle.typedb.studio.test.integration
+
+
+import com.vaticle.typedb.client.api.TypeDBSession
+import com.vaticle.typedb.client.api.TypeDBTransaction
+import com.vaticle.typedb.studio.framework.material.Icon
+import com.vaticle.typedb.studio.state.StudioState
+import com.vaticle.typedb.studio.state.common.util.Label
+import com.vaticle.typedb.studio.test.integration.common.StudioActions
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.clickIcon
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.clickText
+import com.vaticle.typedb.studio.test.integration.data.Paths.SampleGitHubData
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.connectToTypeDB
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.copyFolder
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.createDatabase
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.openProject
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.writeDataInteractively
+import com.vaticle.typedb.studio.test.integration.common.StudioActions.writeSchemaInteractively
+import com.vaticle.typedb.studio.test.integration.common.TypeDBRunners.withTypeDB
+
+import kotlinx.coroutines.runBlocking
+import org.junit.Test
+
+class QueryRunnerTest: IntegrationTest() {
+
+    @Test
+    fun inferNotExplainRegression() {
+        withTypeDB { typeDB ->
+            runBlocking {
+                connectToTypeDB(composeRule, typeDB.address())
+                createDatabase(composeRule, dbName = testID)
+                copyFolder(source = SampleGitHubData.path, destination = testID)
+                openProject(composeRule, projectDirectory = testID)
+                writeSchemaInteractively(composeRule, dbName = testID, SampleGitHubData.schemaFile)
+                writeDataInteractively(composeRule, dbName = testID, SampleGitHubData.dataFile)
+
+                StudioState.client.session.tryOpen(database = testID, TypeDBSession.Type.DATA)
+                StudioActions.delayAndRecompose(composeRule, StudioActions.Delays.NETWORK_IO)
+
+                StudioState.project.current!!.directory.entries.find {
+                    it.name == SampleGitHubData.collaboratorsQueryFile
+                }!!.asFile().tryOpen()
+
+                clickText(composeRule, Label.DATA.lowercase())
+                clickText(composeRule, Label.READ.lowercase())
+                clickText(composeRule, Label.INFER.lowercase())
+
+                clickIcon(composeRule, Icon.RUN, delayMillis = 0)
```

Comment:
Hm. Is eliminating the delay a good idea? This was responsible for highly sporadic test failures previously, right?

The condition "if delay > 0" that has been added to the `clickIcon` function also feels a little arbitrary.

Reply:
Ah, I should have commented this on this.

If we delay when there is nothing to be recomposed, then the recomposition will run forever. This query happens extremely quickly, and as such if we wait for 1.5 seconds before recomposing, there will be nothing to recompose. In line with this, I've introduced an `INSTANT` delay. This reads better for both of the places we add additional logic for.